rllib/core/sarl_trainer.py

```diff
@@ -0,0 +1,34 @@
+class SARLTrainer:
+    def train(self, batch):
+        # this is the function is called by the user to train the model
+        for queue in self.queues:
+            queue.put(batch)
+        results = next(self.training_iterator)
+        self.curr_weights = results[0]["module_weights"]
+        return results
+
+    @staticmethod
+    def compute_loss(batch, fwd_out, **kwargs):
```

Comment:
All these methods are defined here as an abstraction while they are neither an interface api nor are being user directly in this class. It's not clear why so many methods are needed here.  For example how is `compute_loss` gonna be used? should another module call this method directly or should this be called internally when you perform update?